flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SortMergeResultPartition.java

```diff
@@ -455,4 +458,28 @@ public int getNumberOfQueuedBuffers(int targetSubpartition) {
     PartitionedFile getResultFile() {
         return resultFile;
     }
+
+    @VisibleForTesting
+    SortMergeResultPartitionReader getPartitionReader() {
+        return partitionReader;
+    }
+
+    private int[] getRandomSubpartitionOrder(int numSubpartitions) {
```

Comment:
I think it better to use a primitive int array to avoid the unbox overhead when this order array is used, besides, the type that SortBuffer accepts is also primitive int array.

Reply:
I think this is not in critical code path (one call for one ResultPartition only) and thus I do not think it would cause too much overhead. For int[], it could be changed to 

```
List<Integer> list = IntStream.range(0, numSubpartitions).boxed().collect(Collectors.toList());
Collections.shuffle(list);
return list.stream().mapToInt(Integer::intValue).toArray();
```

As a whole, I strongly do not recommend to write a `shuffle` method by ourself.

Reply:
I second with @gaoyunhaii to reused existing libs for this shuffle